README.md

```diff
@@ -3,42 +3,141 @@
 [![Build Status](https://travis-ci.org/BBC-News/crimp.png?branch=master)](https://travis-ci.org/BBC-News/crimp)
 [![Gem Version](https://badge.fury.io/rb/crimp.png)](http://badge.fury.io/rb/crimp)
 
-Creating an md5 hash of a number, string, array, or hash in Ruby
-
-![mighty-boosh-four-way-crimp-o](https://f.cloud.github.com/assets/180050/2148112/b44fd6fa-93de-11e3-9f9a-ad941f069b5c.gif)
-
-Shamelessly copied from [this Stack Overflow
-answer](http://stackoverflow.com/a/6462589/3243663).
+Creates an MD5 hash of a data structure made of numbers, strings, booleans, arrays, sets or hashes.
```

Comment:
I know we flatten the arrays, but this says we create an MD5 hash of arrays. Would we expect these to return the same signature?
```rb
Crimp.signature([])
=> "d41d8cd98f00b204e9800998ecf8427e"
Crimp.signature([[]])
=> "d41d8cd98f00b204e9800998ecf8427e"
```

Reply:
Not sure if an empty array should return the same as an empty hash?
```rb
Crimp.signature({})
=> "d41d8cd98f00b204e9800998ecf8427e"
```

Reply:
Good point. This happens because the idea is to reduce the content to its primitives. so yes, in this case, an empty hash == an empty array...

Reply:
I can easily add a 'byte' to indicate the data structure type:
```ruby
Crimp.to_a([123, 'abc'])
=>  [[[123, "N"], ["abc", "S"]], "A"]
```

Reply:
```ruby
Crimp.to_a({a: 'b'})
=>  [[[[["a", "S"], ["b", "S"]], "A"]], "H"]
```

Reply:
I've added a commit to this PR to address this issue, if is fine I'll update the README and add a couple of tests for edge cases

Reply:
Set will need to be removed from the readme now.